src/harness/testcases/WINNF_FT_S_SCS_testcase.py

```diff
@@ -0,0 +1,170 @@
+#    Copyright 2016 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+import json
+import os
+import re
+import socket
+import urlparse
+
+from OpenSSL import crypto, SSL
+
+import sas
+import sas_testcase
+from util import winnforum_testcase
+
+
+_PEM_RE = re.compile(b"""-----BEGIN CERTIFICATE-----\r?
+.+?\r?
+-----END CERTIFICATE-----\r?\n?""", re.DOTALL)
+
+
+# Keeps this function as module function for easy mocking to adapt to specific
+# SAS implementation.
+def _VerifyCertificate(certificate_content, ca_bundle_content, is_server):
+  """Verifies the given certificate from the given chain store.
+
+  Does verification described in: crypto.load_certificate(crypto.FILETYPE_PEM, match.group(0))
+
+  Args:
+    certificate_content: certificate to validate as raw file content.
+    ca_bundle_content: certificate bundle content to be used as trusted CA.
+    is_server: True if certificate is a server certificate, False for a
+      client certificate.
+  Raises:
+    crypto.X509StoreContextError if certificate not valid
+  """
+  # is_server not used on this generic implementation...
+  certificate = crypto.load_certificate(crypto.FILETYPE_PEM, certificate_content)
+  store = crypto.X509Store()
+  for match in _PEM_RE.finditer(ca_bundle_content):
+    store.add_cert(crypto.load_certificate(crypto.FILETYPE_PEM, match.group(0)))
+  store_ctx = crypto.X509StoreContext(store, certificate)
+  store_ctx.verify_certificate()
+
+
+class SecurityTestcase(sas_testcase.SasTestCase):
+
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  def assertValidClientCertificate(self, certificate_filename,
```

Comment:
Same as for previous comment: these 'asserts' are very specific to the SCS testcase. So either I keep them here or I introduce a new security_util.py for them.